### PR TITLE
Encapsulate  the row rendering logic in a separate component

### PR DESCRIPTION
**Pull request by sherrif10**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Other

I introduced CustomTableRow Component  which Promotes a cleaner and more modular structure by encapsulating the row rendering logic in a CustomTableRow  component. 

Also i did some  code refactory makes it easier to manage the changes in the structure of the table.

Though the initial approach also has an advantage in the way that it makes each table cell individually defined which makes customization little easy than this approach which is more generic. Let me know your suggestions.  cc @denniskigen  @pirupius @jabahum . Thanks.

Screenshot from 2024-01-10 15-02-22

### CODE REVIEW

**Comment by jabahum** [comment]
hey @sherrif10 do you mind fixing the build errors

**Comment by sherrif10** [comment]
Fixed linters cc @jabahum

File: src/completed-list/completed-list.component.tsx [hunk 1]
```diff
@@ -34,11 +34,88 @@ import {
...
+    urgency,
+    dateActivated,
+    patient,
+  } = entry;
+
+  return (
+    <TableRow key={uuid}>
+      <TableCell>
+        <span>{formatDate(parseDate(dateActivated))}</span>
+      </TableCell>
```
Comment (pirupius) [inline src/completed-list/completed-list.component.tsx:89]: add class to make the cell display date on a single line

**Review by pirupius — CHANGES_REQUESTED** [review]
Thanks @sherrif10 just one more tweak before we can get this in

File: src/completed-list/completed-list.component.tsx (hunk 1 above)
Comment (jabahum) [reply src/completed-list/completed-list.component.tsx]: @sherrif10 and @pirupius hopefully the actions button will also be in a single line

**Comment by jabahum** [comment]
@sherrif10 to avoid lint errors just make sure to run yarn verify before pushing to confirm everything is fine

File: src/completed-list/completed-list.scss [hunk 2]
```diff
@@ -229,4 +229,18 @@ title {
...
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+.single-line-action {
+  display: inline-block;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
```
Comment (pirupius) [inline src/completed-list/completed-list.scss:245]: This is duplication. You can change the class name to something more generic and reuse it in both scenarios

Comment (sherrif10) [reply src/completed-list/completed-list.scss]: Thanks for reviews

Comment (pirupius) [reply src/completed-list/completed-list.scss]: @sherrif10 please update the PR description with the latest screenshot

Comment (sherrif10) [reply src/completed-list/completed-list.scss]: @pirupius  i added the lasted screen shot.

Comment (pirupius) [reply src/completed-list/completed-list.scss]: @sherrif10 your screenshot shows the date breaking to the next line. The date should be on a single line.

Comment (pirupius) [reply src/completed-list/completed-list.scss]: @sherrif10 You can use something like this in the class you added

```
white-space: nowrap;
```

File: src/completed-list/completed-list.scss [hunk 3]
```diff
@@ -229,4 +229,11 @@ title {
   @include type.type-style('heading-compact-02');
   color: $text-02;
   margin-bottom: 0.5rem;
-}
\ No newline at end of file
+}
+
+.single-line-content {
+  display: inline-block;
+  white-space: nowrap;
```
Comment (sherrif10) [inline src/completed-list/completed-list.scss:236]: Hey @pirupius . i used it here .

Comment (sherrif10) [reply src/completed-list/completed-list.scss]: Hey @pirupius , Updated the screen shot.

File: src/queue-list/laboratory-queue.scss [hunk 4]
```diff
@@ -210,4 +210,9 @@ title {
   @include type.type-style('heading-compact-02');
   color: $text-02;
   margin-bottom: 0.5rem;
-}
\ No newline at end of file
+}
+
+  .table {
```
Comment (pirupius) [inline src/queue-list/laboratory-queue.scss:215]: Let's change this to something more intentional and also not to conflict with other usages. What do u think of changing it to `.single-line-display`

**Review by pirupius — APPROVED** [review]
Thanks @sherrif10

**Pull request merged**